packages/cozy-doctypes/src/administrativeProcedures/AdministrativeProcedure.js

```diff
@@ -46,6 +46,63 @@ class AdministrativeProcedure extends Document {
 
     return personalData
   }
+  /**
+   * Method to generate a query based on a few rules given by the template
+   * @param {Object} docRules
+   * @param {Object} docRules.rules
+   * @param {int} docRules.count
+   */
+  static async getFilesByRules(docRules) {
+    const { rules, count } = docRules
+    const cozyRules = {
+      trashed: false,
+      type: 'file',
+      ...rules
+    }
+    //Create an index to be order to query and sort
+    await this.cozyClient
+      .collection('io.cozy.files')
+      .createIndex(['metadata.datetime', 'metadata.classification'])
+    //Use the index
+    const files = await this.cozyClient
+      .collection('io.cozy.files')
+      .find(cozyRules, {
+        indexedFields: ['metadata.datetime', 'metadata.classification'],
+        sort: [
+          {
+            'metadata.datetime': 'desc'
+          },
+          {
+            'metadata.classification': 'desc'
+          }
+        ],
+        limit: count ? count : 1
+      })
+
+    return files
+  }
+  /**
+   * This function is used to populate an array based on the order
+   * from the documentsTemplate and set the linked document from the store
+   * @param {Object} docsFromStore
+   * @param {Object} documentsTemplate
+   */
+  static mergeDocsFromStoreAndTemplate(docsFromStore, documentsTemplate) {
```

Comment:
I'm not sure it's the right place in doctypes, as this seems very specific to the procedure app (it takes the store as param).